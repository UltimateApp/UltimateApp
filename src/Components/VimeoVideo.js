import React, { useEffect, useRef, useState } from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';

import { Video } from 'expo-av';
import theme from '../styles/theme.style';

const VimeoVideo = ({ vimeoId, screenWidth }) => {
  const vimeoUrlSource = `https://player.vimeo.com/video/${vimeoId}/config`;
  const videoElem = useRef(null);
  const [isBuffering, setBuffer] = useState(true);

  useEffect(() => {
    fetch(vimeoUrlSource)
      .then(res => res.json())
      .then(res => {
        const videoArray = res.request.files.progressive;
        const videoVimeoQuality = videoArray.find(videoObject => videoObject.quality === '540p');
        if (videoVimeoQuality) {
          return videoVimeoQuality.url;
        }
      })
      .then(url => {
        try {
          videoElem.current.loadAsync({
            uri: url,
          });
        } catch (e) {
          console.log('ERROR Loading Video', e);
        }
      });
  }, [vimeoUrlSource]);

  const playVideoLoaded = () => {
    videoElem.current.setStatusAsync({
      rate: 1.0,
      isMuted: true,
      resizeMode: Video.RESIZE_MODE_CONTAIN,
      shouldPlay: true,
      isLooping: true,
    });
    setBuffer(false);
  };

  const renderBufferIcon = () => {
    if (isBuffering) {
      return (
        <View style={styles.spinnerStyle}>
          <ActivityIndicator animating color={theme.COLOR_SECONDARY} size="large" />
        </View>
      );
    }
  };

  return (
    <View style={styles.videoContainer}>
      {renderBufferIcon()}
      <Video
        ref={videoElem}
        resizeMode={Video.RESIZE_MODE_CONTAIN}
        style={{ width: screenWidth, height: 250 }}
        onLoadStart={() => setBuffer(true)}
        onLoad={playVideoLoaded}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  videoContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  spinnerStyle: {
    position: 'absolute',
  },
});

export default VimeoVideo;
