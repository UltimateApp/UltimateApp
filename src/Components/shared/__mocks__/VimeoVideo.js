import React, { ReactElement } from 'react';

import { View } from 'react-native';

const MockedVimeoVideo = (props): ReactElement => {
  return <View testID="VimeoVideoMock" {...props} />;
};

export default MockedVimeoVideo;
