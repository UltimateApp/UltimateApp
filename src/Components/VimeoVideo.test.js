import React from 'react';
import renderer from 'react-test-renderer';

import VimeoVideo from './VimeoVideo';
import { VIMEO_VIDEO_ID } from '../Fixtures';

describe('<VimeoVideo />', () => {
  it('renders correctly for a video from vimeo', () => {
    const screenWidth = 480;
    const tree = renderer.create(<VimeoVideo vimeoId={VIMEO_VIDEO_ID} screenWidth={screenWidth} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
