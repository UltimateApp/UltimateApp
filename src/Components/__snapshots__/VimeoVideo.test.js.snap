// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VimeoVideo /> renders correctly for a video from vimeo 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#A6A6A6"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "height": 250,
        "overflow": "hidden",
        "width": 480,
      }
    }
  >
    <ViewManagerAdapter_ExpoVideoView
      proxiedProperties={
        Object {
          "onError": [Function],
          "onFullscreenUpdate": [Function],
          "onLoad": [Function],
          "onLoadStart": [Function],
          "onReadyForDisplay": [Function],
          "onStatusUpdate": [Function],
          "resizeMode": "mock",
          "source": undefined,
          "status": Object {},
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
